Include/code.h

```diff
@@ -102,7 +103,7 @@ PyAPI_DATA(PyTypeObject) PyCode_Type;
 
 /* Public interface */
 PyAPI_FUNC(PyCodeObject *) PyCode_New(
-        int, int, int, int, int, PyObject *, PyObject *,
+        int, int, int, int, int, int, PyObject *, PyObject *,
```

Comment:
Adding a required argument to the code type constructor will breaking every library that creates `types.CodeType` instances. Are you sure this is a good idea?

Reply:
Looks like it has already started: https://www.google.com/search?q=TypeError%3A+code%28%29+%22takes+at+least+14+arguments+%2813+given%29%22

Reply:
You missed the end of the story:

```
commit 4a2edc34a405150d0b23ecfdcb401e7cf59f4650
Author: Pablo Galindo <Pablogsal@gmail.com>
Date:   Mon Jul 1 11:35:05 2019 +0100

    bpo-37221: Add PyCode_NewWithPosOnlyArgs to be used internally and set PyCode_New as a compatibility wrapper (GH-13959)
    
    Add PyCode_NewEx to be used internally and set PyCode_New as a compatibility wrapper
```

Reply:
Also, `types.CodeType`'s constructor is considered private. We now even have this warning in the docs (https://docs.python.org/3.8/library/types.html#standard-interpreter-types):

> If you instantiate any of these types, note that signatures may vary between Python versions.

Reply:
Thank you for the explanation!
I hope in future we'll have a more stable API for dynamic function generation.

Reply:
This breaks `pickle.loads` for functions constructed as
```python
def constant_fn(val: float) -> Callable:
    def func(_):
        return val
    
    return func
fun = constant_fn(1.)
```
and pickled using `cloudpickle` under Python 3.7; these can't be loaded back up under Python 3.8.

Reply:
> Should this be addressed downstream in cloudpickle ?

Yes, they need to update whatever code they are using to create code objects as code objects in Python3.8 require an extra field. Please, check these sections in the docs:

https://docs.python.org/3/library/types.html#standard-interpreter-types

> If you instantiate any of these types, note that signatures may vary between Python versions.

Also, check this new C-API function:

https://docs.python.org/3/c-api/code.html#c.PyCode_NewWithPosOnlyArgs

Reply:
Alright, I'll file an issue there. Thanks.